src/sentry/snuba/metrics/extraction.py

```diff
@@ -182,11 +201,39 @@ def is_on_demand_snuba_query(snuba_query: SnubaQuery) -> bool:
 
 
 def is_on_demand_query(dataset: Optional[Union[str, Dataset]], query: Optional[str]) -> bool:
-    """Returns ``True`` if the dataset and query combination can't be supported by standard metrics."""
+    """Returns ``True`` if the dataset is performance metrics and query contains non-standard search fields."""
+
     if not dataset or not query:
         return False
 
-    return Dataset(dataset) == Dataset.PerformanceMetrics and "transaction.duration" in query
+    return Dataset(dataset) == Dataset.PerformanceMetrics and _contains_on_demand_search_filters(
+        query
+    )
+
+
+def _contains_on_demand_search_filters(query: str) -> bool:
+    """Recursively checks if the query contains any non-standard search filters."""
+    try:
+        tokens = event_search.parse_search_query(query)
+
+        for token in tokens:
+            if _is_on_demand_search_filter(token):
+                return True
+
+        return False
+    except InvalidSearchQuery as e:
+        logger.error(f"Failed to parse search query: {query}. Error: {e}")
+        return False
+
+
+def _is_on_demand_search_filter(token: QueryToken) -> bool:
+    if isinstance(token, SearchFilter):
+        return token.key.name not in _STANDARD_METRIC_FIELDS
```

Comment:
Alert queries always have `event.type:transaction` in them, but they can still be compatible with the standard metrics. In ``, we remove this component from the query, but `is_on_demand_query` can be called outside of that. 

I'm not sure whether we should primarily focus on the dataset or the `event.type` filter to determine whether this fundamentally is a performance query. However, this filter should not trigger on-demand metrics per se. 

The simplest solution would probably be to add `event.type` to `_STANDARD_METRIC_FIELDS`.

Reply:
Added `event.type` to `_STANDARD_METRIC_FIELDS` for now. We can revisit it later